### ISSUE TITLE
Waiting for request with cy.wait when alias name matches tag name breakes cy.get command (interception becomes subject)

### ISSUE THREAD

**Issue by shwarcu** [cypress-io/cypress#24740]
### Current behavior

I created repository with reproduction 
https://github.com/shwarcu/cypress-alias-bug-repro

When few conditions are met, `cy.get` command is returning wrong subject (intercepted request instead of DOM element).
It behaves like `cy.get('@aliasname')` instead looking for an actual element. Here are conditions:
- alias has the same name as element we lookup (in reproduction I used `video` alias name and element)
- request is intercepted
- request is awaited with `cy.wait`

Example test that fails

```ts
  it('fails when using get command after previously waiting for alias with the same name', () => {
    cy.intercept(DEFAULT_VIDEO_URL, (request) => {
      request.redirect(OTHER_VIDEO_URL);
    }).as('video');
    cy.visit('');

    cy.wait('@video');


    cy.get('video').should('have.attr', 'src'); // here subject is not video element, but interception object
  });
```

<img width="1723" alt="Screenshot 2022-11-19 at 15 25 07" src="https://user-images.githubusercontent.com/58426925/202855613-04f98b42-658e-46ab-a322-63afa3adb4c5.png">

### Desired behavior

`cy.get('video')` should not behave like `cy.get('@video')`. It should yield video element.

### Test code to reproduce

https://github.com/shwarcu/cypress-alias-bug-repro

### Cypress Version

11.1.0

### Node version

v16.13.0

### Operating System

MacOS 13.0.1 / Manjaro GNOME 43.1

**Comment by ryanthemanuel** [comment]
Hi @shwarcu. Thank you for logging this issue. The screenshot you listed shows `get @video`. Are you sure your `cy.get` is referencing `video` and not `@video`. I couldn't get this to fail with something like `@body` and `body`

**Comment by shwarcu** [comment]
Hi @ryanthemanuel , I am literally running the test I presented in the test with reproduction https://github.com/shwarcu/cypress-alias-bug-repro/blob/main/cypress/e2e/example.spec.cy.ts
Failling command is
```ts
cy.get('video').should('have.attr', 'src');
```
![Screenshot 2022-11-21 at 23 21 15](https://user-images.githubusercontent.com/58426925/203169837-55ad1a2a-a463-4362-bcd6-624054371c9a.png)


Not sure if I explained the problem clearly enough so I will try again:
under these particular conditions I have listed above and presented in reproduction code, 
this command

```ts
cy.get('video').should('have.attr', 'src');
```

will work as if I did

```ts
cy.get('@video').should('have.attr', 'src');
```

Meaning that `cy.get('video')` yields interception object, instead of `video` element from the page.

Official documentation on that topic https://docs.cypress.io/guides/core-concepts/variables-and-aliases#Intercepts
explains that we use `@aliasName` to inside `cy.get` in order to access aliased interception.

So I expect that if I don't use `@`, it won't yield my aliased subject.


**Edit: I think I have missed one thing in the issue description - intercepted request should be modified - in my case it's redirection.** But this case is included in reproduction spec.

**Comment by ryanthemanuel** [comment]
Apologies @shwarcu. I think I completely skipped over your link to your reproduction and consequently was trying to set up something myself. I can confirm that your reproduction at https://github.com/shwarcu/cypress-alias-bug-repro/blob/main/cypress/e2e/example.spec.cy.ts reproduces this issue and I will route this to the e2e team for prioritization.

**Comment by shwarcu** [comment]
No problem! Thanks a lot for taking a look at it so quickly 🎉 Let me know once there is anything else I can do to help with that bug 🐛

**Comment by BlueWinds** [comment]
Interesting, and very unusual - thank you for the repro!

This appears to be fixed in Cypress 12.0.0, which should be coming out later this week. You can try it out with the pre-release binaries if you like (https://github.com/cypress-io/cypress/commit/2552bdda9c8d8a30fa4c570f561f619ccb1cb8b3 is the latest as of this writing).

There appears to be some additional conditions to trigger the bug. This test:

```
  it.only('gets confused', () => {
    cy.intercept(/users/, {}).as('div')

    cy.visit('')
    cy.window().then((win) => {
      return win.Cypress.$.get('/users')
    })

    cy.wait('@div')

    cy.get('div').should('have.attr', 'id')
  })
```

passes in Cy 11.1.0 - `.get('div')` is working as expected. I'm not going to dive too deep into it, since the issue seems to be resolved in Cy12 already.

I'm going to close this, but please feel free to ping me or comment here again if you're still seeing issues in Cy12. I'm the one that rewrote our aliasing logic recently, so if there are problems I always want to hear about them!
